docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -319,6 +320,17 @@ type LocalObjectTemplate struct {
   - `spec.topology.workers.machineDeployments[i].name` field must be unique within a Cluster
   - A set of worker nodes can be added to or removed from the `spec.topology.workers.machineDeployments` list.
 
+##### ClusterClass compatibility
+There are cases where we must consider whether two ClusterClasses are compatible:
+1. Where a user chooses to replace an existing ClusterClass `cluster.spec.topology.class` with a new ClusterClass
+2. Where a user updates a ClusterClass currently in use by a Cluster
```

Comment:
Note (implementation detail for a later stage): In order to make this possible, the rules currently implemented in the validation web hook should be relax